frontend/catalyst/pennylane_extensions.py

```diff
@@ -16,28 +16,32 @@
 while using :func:`~.qjit`.
 """
 
-import uuid
 import functools
 import numbers
+import uuid
 
 import jax
 import jax.numpy as jnp
+import pennylane as qml
 from jax import ShapedArray
-from jax.tree_util import tree_flatten, tree_unflatten, treedef_is_leaf
-from jax.linear_util import wrap_init
-from jax._src.lax.control_flow import _initial_style_jaxpr, _initial_style_jaxprs_with_common_consts
+from jax._src.lax.control_flow import (
+    _initial_style_jaxpr,
+    _initial_style_jaxprs_with_common_consts,
+)
 from jax._src.lax.lax import _abstractify
-
-import pennylane as qml
-from pennylane.operation import Operation, Wires, AnyWires
+from jax.linear_util import wrap_init
+from jax.tree_util import tree_flatten, tree_unflatten, treedef_is_leaf
 from pennylane.measurements import MidMeasureMP
+from pennylane.operation import AnyWires, Operation, Wires
 
+import catalyst
```

Comment:
```suggestion
from catalyst.compilation_pipelines import QJIT
```

Reply:
Doesn't work unfortunately (circular imports).

Reply:
Ah.. right! We may need to redesign it now to get rid of this circular imports. What would you think?

Reply:
I don't think a redesign is needed for this PR. The problem is just an `isinstance` check, for which there are two solutions:
- Use strings to check the class rather the class object itself.
- Don't import the class directly.

Out of the two, I think the latter works just fine.

Reply:
I don't think a redesign is needed for this PR. The problem is just an `isinstance` check, for which there are two solutions:
- Use strings to check the class rather the class object itself.
- Don't import the class directly.

Out of the two, I think the latter works just fine.